plugins/in_zmq/in_zmq.c

```diff
@@ -0,0 +1,296 @@
+         * time, and the second is a MAP of keys with their values.
+         */
+        msgpack_pack_array(&mp_pck, 2);
+        flb_pack_time_now(&mp_pck);
+
+        /* We then store the 3 parts - "topic", "key", and "payload". */
+        msgpack_pack_map(&mp_pck, 3);
+
+        frame = zmsg_pop(zmsg);
+        if (unlikely(frame == NULL)) {
+            flb_warn("[in_zmq] dropping message with missing frame 1 "
+                     "(%d)", num_frames);
+            msgpack_sbuffer_destroy(&mp_sbuf);
+            zmsg_destroy(&zmsg);
+            continue;
+        }
+        msgpack_pack_str(&mp_pck, 5);
+        msgpack_pack_str_body(&mp_pck, "topic", 5);
+        msgpack_pack_str(&mp_pck, zframe_size(frame));
+        msgpack_pack_str_body(&mp_pck, zframe_data(frame), zframe_size(frame));
+        zframe_destroy(&frame);
+
+        frame = zmsg_pop(zmsg);
+        if (unlikely(frame == NULL)) {
+            flb_warn("[in_zmq] dropping message with missing frame 2 "
+                     "(%d)", num_frames);
+            msgpack_sbuffer_destroy(&mp_sbuf);
+            zmsg_destroy(&zmsg);
+            continue;
+        }
+        msgpack_pack_str(&mp_pck, 3);
+        msgpack_pack_str_body(&mp_pck, "key", 3);
+        msgpack_pack_str(&mp_pck, zframe_size(frame));
+        msgpack_pack_str_body(&mp_pck, zframe_data(frame), zframe_size(frame));
+        zframe_destroy(&frame);
+
+        frame = zmsg_pop(zmsg);
+        if (unlikely(frame == NULL)) {
+            flb_warn("[in_zmq] dropping message with missing frame 3 "
+                     "(%d)", num_frames);
+            msgpack_sbuffer_destroy(&mp_sbuf);
+            zmsg_destroy(&zmsg);
+            continue;
+        }
+        msgpack_pack_str(&mp_pck, 7);
+        msgpack_pack_str_body(&mp_pck, "payload", 7);
+        msgpack_pack_bin(&mp_pck, zframe_size(frame));
+        msgpack_pack_bin_body(&mp_pck, zframe_data(frame), zframe_size(frame));
+        zframe_destroy(&frame);
+
+        ret = flb_input_chunk_append_raw(in, NULL, 0, mp_sbuf.data,
+                                         mp_sbuf.size);
+        if (unlikely(ret < 0))
+            flb_warn("[in_zmq] flb_input_chunk_append_raw failed of size %u",
+                     mp_sbuf.size);
+        msgpack_sbuffer_destroy(&mp_sbuf);
+
+        zevents = zsock_events(ctx->zmq_pull_socket);
+    }
+
+    return 0;
+}
+
+static void in_zmq_pause(void *data, struct flb_config *config)
+{
+    struct flb_in_zmq_ctx *ctx = data;
+    flb_debug("[in_zmq] pausing endpoint %s on fd %d", ctx->zmq_endpoint,
+              ctx->ul_fd);
+    flb_input_collector_pause(ctx->ul_fd, ctx->i_ins);
+}
+
+static void in_zmq_resume(void *data, struct flb_config *config)
+{
+    struct flb_in_zmq_ctx *ctx = data;
+    flb_debug("[in_zmq] resuming endpoint %s on fd %d", ctx->zmq_endpoint,
+              ctx->ul_fd);
+    flb_input_collector_resume(ctx->ul_fd, ctx->i_ins);
```

Comment:
This is not collect. `flb_input_collector_*` expects the identifier
returned by `flb_input_set_collector_event()` below.